src/Akeneo/Pim/Enrichment/Product/back/Application/Applier/AssociationUserIntentCollectionApplier.php

```diff
@@ -124,49 +83,110 @@ public function getSupportedUserIntents(): array
-        ) {
-            return $normalizedAssociations[$associationUserIntent->associationType()]['product_models'] ??
-                $product
+                    ?->map(fn (ProductInterface $product): string => $product->getIdentifier())?->toArray() ?? [],
+            self::PRODUCT_MODELS => $product
                     ->getAssociatedProductModels($associationUserIntent->associationType())
-                    ?->map(fn (ProductModelInterface $productModel): string => $productModel->getIdentifier())?->toArray() ?? [];
-        }
+                    ?->map(fn (ProductModelInterface $productModel): string => $productModel->getIdentifier())?->toArray() ?? [],
+            self::GROUPS => $product
+                    ->getAssociatedGroups($associationUserIntent->associationType())
+                    ?->map(fn (GroupInterface $group): string => $group->getCode())?->toArray() ?? [],
+            default => throw new \LogicException('Not implemented')
+        };
 
-        throw new \LogicException('Not implemented');
+        return $normalizedAssociations[$associationUserIntent->associationType()][$entityType] ?? $values ?? [];
     }
 
     private function getAssociationEntityType(AssociationUserIntent $userIntent): string
     {
-        if (
-            $userIntent instanceof AssociateProducts
-            || $userIntent instanceof DissociateProducts
-            || $userIntent instanceof ReplaceAssociatedProducts
-        ) {
-            return 'products';
-        } elseif (
-            $userIntent instanceof AssociateProductModels
-            || $userIntent instanceof DissociateProductModels
-            || $userIntent instanceof ReplaceAssociatedProductModels
-        ) {
-            return 'product_models';
+        return match($userIntent::class) {
+            AssociateProducts::class, DissociateProducts::class, ReplaceAssociatedProducts::class => self::PRODUCTS,
+            AssociateProductModels::class, DissociateProductModels::class, ReplaceAssociatedProductModels::class => self::PRODUCT_MODELS,
+            AssociateGroups::class, DissociateGroups::class, ReplaceAssociatedGroups::class => self::GROUPS,
+            default => throw new \LogicException('Level does not exists')
+        };
+    }
+
+    /**
+     * @return array<string>
+     */
+    private function userIntentEntityAssociations(AssociationUserIntent $associationUserIntent, string $entityType): array
+    {
+        if ($entityType === self::PRODUCTS && \method_exists($associationUserIntent, 'productIdentifiers')) {
+            return $associationUserIntent->productIdentifiers();
+        } elseif ($entityType === self::PRODUCT_MODELS && \method_exists($associationUserIntent, 'productModelCodes')) {
+            return $associationUserIntent->productModelCodes();
+        } elseif ($entityType === self::GROUPS && \method_exists($associationUserIntent, 'groupCodes')) {
+            return $associationUserIntent->groupCodes();
         }
```

Comment:
```suggestion
        if ($entityType === self::PRODUCTS) {
            return $associationUserIntent->productIdentifiers();
        } elseif ($entityType === self::PRODUCT_MODELS) {
            return $associationUserIntent->productModelCodes();
        } elseif ($entityType === self::GROUPS) {
            return $associationUserIntent->groupCodes();
        }
```

You can remove the method_exist. The code will also fail if the method does not exist

Reply:
Without them phpStan throws an error, should I add php_stan ignore or keep the methods?

Reply:
Ok keep like that